system/jlib/jptree.cpp

```diff
@@ -7961,3 +7961,262 @@ jlib_decl IPropertyTree * loadConfiguration(const char * defaultYaml, const char
+                iEvent->endNode(tagname, event.data.scalar.length, (const void *)event.data.scalar.value, false, parser.offset);
+                break;
+            case YAML_ALIAS_EVENT: //reference to an anchor, ignore for now
+                iEvent->beginNode(tagname, parser.offset);
+                iEvent->endNode(tagname, 0, nullptr, false, parser.offset);
+                break;
+            case YAML_SEQUENCE_END_EVENT: //done
+                break;
+            case YAML_NO_EVENT:
+            case YAML_MAPPING_END_EVENT:
+            case YAML_STREAM_START_EVENT:
+            case YAML_STREAM_END_EVENT:
+            case YAML_DOCUMENT_START_EVENT:
+            case YAML_DOCUMENT_END_EVENT:
+            default:
+                //shouldn't be here
+                break;
+            }
+
+            yaml_event_delete(&event);
+        }
+    }
+    virtual void loadMap(const char *tagname)
+    {
+        StringBuffer content;
+        if (tagname && *tagname)
+            iEvent->beginNode(tagname, parser.offset);
+
+        yaml_event_t event;
+        yaml_event_type_t eventType = YAML_NO_EVENT;
+
+        while (eventType!=YAML_MAPPING_END_EVENT)
+        {
+            eventType = nextEvent(event, YAML_MAPPING_END_EVENT, YAML_SCALAR_EVENT, "expected map to start with scalar name");
+            if (eventType==YAML_MAPPING_END_EVENT)
+                continue;
+            StringBuffer name(event.data.scalar.length, (const char *)event.data.scalar.value);
+            yaml_event_delete(&event);
+
+            eventType = nextEvent(event);
+
+            switch (eventType)
+            {
+            case YAML_MAPPING_START_EVENT: //child map
+                loadMap(name);
+                break;
+            case YAML_SEQUENCE_START_EVENT:
+                loadSequence(name);
+                break;
+            case YAML_SCALAR_EVENT:
+            {
+                //!el or !element will be our local tag (custom schema type) for an element
+                //ptree toYAML should set this for element scalars, and parent text content
+                if (event.data.scalar.tag && (streq((const char *)event.data.scalar.tag, "!el") || streq((const char *)event.data.scalar.tag, "!element")))
+                {
+                    if (streq(name, "^")) //text content of parent node
+                        content.append(event.data.scalar.length, (const char *) event.data.scalar.value);
+                    else
+                    {
+                        iEvent->beginNode(name, parser.offset);
+                        iEvent->endNode(name, event.data.scalar.length, (const void *) event.data.scalar.value, false, parser.offset);
+                    }
+                }
+                else //by default all named scalars are ptree attributes
+                {
+                    name.insert(0, '@');
+                    StringAttr value((const char *) event.data.scalar.value, event.data.scalar.length);
```

Comment:
minor: If typical length is < 16 characters then a local StringBuffer may be more efficient.  Even more efficient would be a way of avoiding creating a temporary.

Reply:
A bit unfortunate that setAttribute() requires the '@' since it is implied.  I may be able to avoid the insert.

Reply:
actually value does seem to be null terminated, so should be able to use directly.